generator/views/metric_definitions_view.py

```diff
@@ -0,0 +1,207 @@
+              FROM
+                {
+                    MetricsConfigLoader.configs.get_data_source_sql(
+                        data_source_name,
+                        self.namespace
+                    ).format(dataset=self.namespace)
+                }
+              WHERE {data_source_definition.submission_date_column} BETWEEN
+                SAFE_CAST({{% date_start {base_view_name}.date %}} AS DATE) AND
+                SAFE_CAST({{% date_end {base_view_name}.date %}} AS DATE)
+              GROUP BY
+                client_id,
+                submission_date
+            """
+        }
+        view_defn["dimensions"] = self.get_dimensions()
+        view_defn["dimension_groups"] = self.get_dimension_groups()
+        view_defn["measures"] = []
+
+        # Custom filter injected into the derived table SQL to filter on the date partition.
+        # All metric definition views reference the date filter from the 'base_view'.
+        # This allows us to have a single filter in the explore that gets applied to all the views.
+        view_defn["filters"] = [
+            {
+                "name": "date",
+                "type": "date",
+                "description": "Date Range",
+            }
+        ]
+        view_defn["sets"] = self._get_sets()
+
+        return {"views": [view_defn]}
+
+    def get_dimensions(
+        self, _bq_client=None, _table=None, _v1_name: Optional[str] = None
+    ) -> List[Dict[str, Any]]:
+        """Get the set of dimensions for this view based on the metric definitions in metric-hub."""
+        namespace_definitions = MetricsConfigLoader.configs.get_platform_definitions(
+            self.namespace
+        )
+        metric_definitions = namespace_definitions.metrics.definitions
+        data_source_name = re.sub("^metric_definitions_", "", self.name)
+
+        # Each view has a client_id dimension.
+        # For every metric definition view, except the base view, this dimension won't be exposed.
+        # So the following logic only gets run for the base view.
+        # The client_id column can be overridden in metric-hub, or when joining metric definition
+        # view some client_ids might not be available in the base view, but in the view that gets joined.
+        # This would result in the selected client_id being set to null (since we always select the client_id)
+        # from the base view.
+        # The following logic selects the first available client_id from any view that is part of the join.
+        joined_client_id_columns = "SAFE_CAST(${TABLE}.client_id AS STRING)"
+        for data_source in MetricsConfigLoader.data_sources_of_namespace(
+            self.namespace
+        ):
+            # for any view that is part of the current query, use the first available client_id
+            joined_client_id_columns += f"""
+                {{% if  metric_definitions_{data_source}._in_query %}}
```

Comment:
```suggestion
                {{%- if  metric_definitions_{data_source}._in_query %}}
                , SAFE_CAST(metric_definitions_{data_source}.client_id AS STRING)
                {{%- endif -%}}
```
not sure if the `-` would survive, but currently the formating of the sql is ... not pretty

Reply:
you changed it for the timestamp in line 161 but not for the customer_id :)